#!/bin/sh -e
## versioned_session.dpatch by Eduard Bloch <blade@debian.org>
##
## DP: Custom name extension of all icewm programs executed from
## icewm-session and hotkeys for theme submenues

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad icewm-1.2.23~/src/apppstatus.cc icewm-1.2.23/src/apppstatus.cc
--- icewm-1.2.23~/src/apppstatus.cc	2005-08-14 20:33:08.000000000 +0200
+++ icewm-1.2.23/src/apppstatus.cc	2006-01-13 14:49:50.000000000 +0100
@@ -422,6 +422,21 @@
     }
     return false;
 #else
+#ifdef linux
+    FILE *fp;
+    fp = fopen("/proc/net/if_inet6", "r");
+    if ( fp ){
+        char buf[32];
+        while ( fscanf(fp, "%*s %*d %*d %*d %*d %s\n",  &buf) == 1 ){
+            if ( strncmp(fNetDev, buf, strlen(fNetDev))==0 ){
+                fclose(fp);
+                return true;
+            }
+        }
+        fclose(fp);
+        return false;
+    }
+#endif
     char buffer[32 * sizeof(struct ifreq)];
     struct ifconf ifc;
     struct ifreq *ifr;
